### PR TITLE
[SYCL] Enable Auto GRF by default on PVC on Linux

### PR DESCRIPTION
**Pull request by sarnex**
Enable auto GRF mode by default on PVC on Linux. We are not doing this for Windows because the driver doesn't support it.

We have run extensive internal testing and are convinced this will not cause performance regressions, and will improve performance in some cases.

You can revert to the previous behavior with `-ftarget-register-alloc-mode=pvc:default`, and we will monitor performance tracking after this and react as necessary.

### CODE REVIEW

**Review by RaviNarayanaswamy — APPROVED** [review]

**Review by rakeshkrishnaiyer — APPROVED** [review]
lgtm

**Review by xtian-github — APPROVED** [review]
LGTM.

File: sycl/test-e2e/KernelAndProgram/target_register_alloc_mode.cpp [hunk 1]
```diff
@@ -2,12 +2,14 @@
...
 
-// RUN: env SYCL_PI_TRACE=-1 %{run} %t_with.out 2>&1 | FileCheck --check-prefix=CHECK-WITH %s
-// RUN: env SYCL_PI_TRACE=-1 %{run} %t_without.out 2>&1 | FileCheck --implicit-check-not=-ze-intel-enable-auto-large-GRF-mode %s
+// RUN: env SYCL_PI_TRACE=-1 %{run} %t_with.out 2>&1 | FileCheck --check-prefix=CHECK-OPT %s
+// RUN: env SYCL_PI_TRACE=-1 %{run} %t_without.out 2>&1 | FileCheck %if system-windows %{ --implicit-check-not=-ze-intel-enable-auto-large-GRF-mode %} %else %{ --check-prefix=CHECK-OPT %} %s
+// RUN: env SYCL_PI_TRACE=-1 %{run} %t_default.out 2>&1 | FileCheck --implicit-check-not=-ze-intel-enable-auto-large-GRF-mode %s
 
-// CHECK-WITH: ---> piProgramBuild(
-// CHECK-WITH: -ze-intel-enable-auto-large-GRF-mode
+// CHECK-OPT: ---> piProgramBuild(
```
Comment (KseniyaTikhomirova) [inline sycl/test-e2e/KernelAndProgram/target_register_alloc_mode.cpp:11]: could you please clarify what is the difference between check-with and check-opt? I haven't find any related stuff in FileCheck impl for both of them.

Comment (sarnex) [reply sycl/test-e2e/KernelAndProgram/target_register_alloc_mode.cpp:11]: sorry, it's just a custom prefix i'm using, it's not a special option inside FileCheck like CHECK-COUNT or something. you can see i set check-prefix in FileCheck calls at the top

Comment (KseniyaTikhomirova) [reply sycl/test-e2e/KernelAndProgram/target_register_alloc_mode.cpp:11]: yes, I noticed your change of command lines. So it is just a custom prefix that is checked like **must-to-have** in output content, right?

Comment (sarnex) [reply sycl/test-e2e/KernelAndProgram/target_register_alloc_mode.cpp:11]: yes it's the exact same behavior is `CHECK`, it is must have or it will fail

Comment (sarnex) [reply sycl/test-e2e/KernelAndProgram/target_register_alloc_mode.cpp:11]: thanks for the review!

**Comment by sarnex** [comment]
@mdtoguchi Sorry, mind taking a look at this bad boy? Thanks

File: clang/test/Driver/sycl-ftarget-register-alloc-mode.cpp [hunk 2]
```diff
@@ -16,6 +16,10 @@
 // RUN:    -fsycl-targets=spir64_gen -ftarget-register-alloc-mode=pvc:default %s 2>&1 \
 // RUN:   | FileCheck -check-prefix=DEFAULT_AOT %s
 
+// RUN: %clang -### -fsycl \
+// RUN:    -fsycl-targets=spir64_gen %s 2>&1 \
+// RUN:   | FileCheck %if system-windows %{ -check-prefix=DEFAULT_AOT %} %else %{ -check-prefix=AUTO_AOT %} %s
```
Comment (mdtoguchi) [inline clang/test/Driver/sycl-ftarget-register-alloc-mode.cpp:21]: I've never seen the if usage like this - pretty neat.

**Review by mdtoguchi — APPROVED** [review]

File: clang/test/Driver/sycl-ftarget-register-alloc-mode.cpp (hunk 2 above)
Comment (sarnex) [reply clang/test/Driver/sycl-ftarget-register-alloc-mode.cpp:21]: yeah i learned this from @aelovikov-intel i think, it's really useful for cases like this

**Pull request merged**